lib/metasploit/framework/login_scanner/winrm.rb

```diff
@@ -39,13 +40,35 @@ def set_sane_defaults
           super
         end
 
+        def parse_auth_methods(resp)
+          return [] unless resp and resp.code == 401
+          methods = []
+          methods << "Negotiate" if resp.headers['WWW-Authenticate'].include? "Negotiate"
+          methods << "Kerberos" if resp.headers['WWW-Authenticate'].include? "Kerberos"
+          methods << "Basic" if resp.headers['WWW-Authenticate'].include? "Basic"
+          return methods
+        end
+
         # send an HTTP request that WinRM would consider as valid  (SOAP XML in the message matching the XML schema definition)
         def send_request(opts)
+          opts['authenticate'] = false
+          allowed_auth_methods = parse_auth_methods(super(opts))
+          opts['authenticate'] = true
```

Comment:
This side effect was unexpected for me, maybe something like this is cleaner:

```suggestion
          allowed_auth_methods = parse_auth_methods(super(opts.merge({ 'authenticate' => true }))
```

But I don't have the full picture here to know if this is the best approach overall just yet